### ISSUE TITLE
Bug: Movie posters are falling back to "no poster" view

### ISSUE THREAD

**Issue by jsaelhof** [jsaelhof/md4k#96]
Movie posters are falling back to "no poster" view when changing the sort direction or updating the date in the watched list

**Comment by jsaelhof** [comment]
Some interesting notes here:

- This seems to happen intermittently. For a while I thought it wasn't actually happening but then suddenly it appeared when I switched to the second list (not the default loaded list). I'm not sure if it was happening there because that list was much shorter than the main list? 

- I think the issue has something to do with components that are reconciled by key. The posters are in a map so they need a key but when i remove the key (causing a lint error), the bug goes away. This is because the components are remounting instead of being re-used which causes them to re-evaluate the intersection. 

- Trying to console log what is happening is hard. I made a list with only 1 movie to replicate the issue and minimize logs. What I'm seeing is weird. Logs about renders are not consistent...sometimes they appear and sometimes they don't. I've also observed that flipping from descending to ascending sort direction would cause the logs to print but flipping from ascending to descending would not (this part seemed consistent). In either case, the same thing happens so it didn't make much sense.

- I found that the Rooks implementation of useInViewRef (which has an almost identical interface to my hook) worked consistently. I didn't really want to rely on Rooks for additional hooks but this is a really important feature and I need it to work consistently and I'm not sure how to debug this further.
